tests/test_bar.py

```diff
@@ -76,16 +104,48 @@ def test_df_from_u_kln_compare_with_pymbar_bar(sigma):
     assert df == pytest.approx(df_ref, rel=0.05, abs=0.01)
 
 
+def test_df_and_err_from_u_kln_partial_overlap(partial_overlap_uniform_ukln_example):
+    u_kln = partial_overlap_uniform_ukln_example
+
+    w_F, w_R = works_from_ukln(u_kln)
+
+    # this example has some infinite work values
+    assert np.any(np.isinf(w_F)) or np.any(np.isinf(w_R))
+
+    # pymbar.BAR warns and returns zero for df and uncertainty if inf is present in either input
+    assert pymbar.BAR(w_F, w_R) == (0.0, 0.0)
+
+    df, df_err = df_and_err_from_u_kln(u_kln)
+    assert np.isfinite(df) and df != 0.0
```

Comment:
nit: could strengthen this assertion about `df`, as df_ref is I think in this example

Reply:
Done in https://github.com/proteneer/timemachine/pull/1098/commits/d3120b8e1bdb80bf4dbe7ff32aa86ae12ef9243b